gupb/controller/claret_wolf.py

```diff
@@ -238,11 +241,14 @@ def update_weapons_knowledge(self, knowledge: characters.ChampionKnowledge) -> N
         weapons = {k: v.loot.name for k, v in weapons.items()}
         for weapon in weapons.items():
             self.weapons_knowledge[weapon[0]] = weapon[1]
+        self.dynamic_obsticles = self.weapons_knowledge.copy()
```

Comment:
it is a shallow copy so it might be worthy to use deepcopy() to have deep copy of self.weapons_knowledge ?